libsolidity/formal/BMC.cpp

```diff
@@ -258,8 +254,7 @@ bool BMC::visit(Conditional const& _op)
 bool BMC::visit(WhileStatement const& _node)
 {
 	auto indicesBeforeLoop = copyVariableIndices();
-	auto touchedVars = touchedVariables(_node);
-	m_context.resetVariables(touchedVars);
+	m_context.resetVariables(touchedVariables(_node));
```

Comment:
Do we need `touchedVariables(_node)` here or is there another way, maybe using `visitBranch` on the loop body? I'm wondering if by getting rid of `touchedVariables` we could also get rid of `VariableUsage` entirely.

Reply:
I don't know. Do you remember why these variables need to be reset?

Reply:
I don't see a way around this at the moment. To me it makes sense that you reset the touched variables, to analyze the loop as if it could be any iteration.